manager/controllers/app/plotter_generator.go

```diff
@@ -118,18 +118,21 @@ func (p *PlotterGenerator) GetCopyDestination(item *DataInfo, destinationInterfa
 	}, nil
 }
 
-func (p *PlotterGenerator) getAssetDataStore(item *DataInfo) *app.DataStore {
-	vaultMap := make(map[string]app.Vault)
+func (p *PlotterGenerator) getAssetDataStore(item *DataInfo, flow taxonomy.DataFlow, vaultMap map[string]app.Vault) *app.DataStore {
+	if vaultMap == nil {
+		// create a new map in case it does not exist already
+		vaultMap = make(map[string]app.Vault)
+	}
 	if utils.IsVaultEnabled() {
 		// Set the value received from the catalog connector.
 		vaultSecretPath := item.DataDetails.Credentials
-		vaultMap[string(taxonomy.ReadFlow)] = app.Vault{
+		vaultMap[string(flow)] = app.Vault{
```

Comment:
How are these changes related to renaming source/dest?
These changes are problematic, for example, there are no vault credentials for Copy flow but only for Read and Write.

Reply:
@shlomitk1 the changes are taken from #1357. we can leave it out from this pr for now. Thanks